bigbluebutton/bigbluebutton
# build: relocate config for bbb-gql-server and middleware

### What does this PR do?



### Closes Issue(s)

Closes # none
Related: https://github.com/bigbluebutton/bigbluebutton/pull/20904

### Motivation

Having the configurations file in `/etc/default` (and then editing them to place a custom secret) means that subsequent upgrades of the `bbb-graphql-*` package the configuration file will be preserved and new file will be created with `.dpkg-dist` ending. Therefore no new configuration can be automatically applied.

### How to test

Install, modify `/usr/share/bbb-graphql-server/bbb-graphql-server.env` (change a value), install a newer version of `bbb-graphql-server`. Inspect `/usr/share/bbb-graphql-server/` -- there should be no files ending in `.dpkg-dist` and the .env file should be overridden -- the change you made earlier should be gone.

### More

Next step (@gustavotrott perhaps?) would be to add support for a file (for each of -server and -middleware) such that:
`/etc/bigbluebutton/bbb-graphql-server.env` (or similar) with content ABC=10 would replace the ABC value from `/usr/share/bbb-graphql-server/bbb-graphql-server.env`
The same for -middleware.

These two new files should be documented in https://github.com/bigbluebutton/bigbluebutton/blob/v3.0.x-release/docs/docs/administration/configuration-files.md?plain=1#L22
